packages/server/lib/clients/oauth2.client.ts

```diff
@@ -49,7 +49,15 @@ export function getSimpleOAuth2ClientConfig(providerConfig: ProviderConfig, temp
 
 export async function getFreshOAuth2Credentials(connection: Connection, config: ProviderConfig, template: ProviderTemplateOAuth2): Promise<OAuth2Credentials> {
     let credentials = connection.credentials as OAuth2Credentials;
-    const client = new AuthorizationCode(getSimpleOAuth2ClientConfig(config, template, connection.connection_config));
+    const simpleOAuth2ClientConfig = getSimpleOAuth2ClientConfig(config, template, connection.connection_config);
+    if (template.token_request_auth_method === 'basic') {
```

Comment:
Isn't this redundant with https://github.com/NangoHQ/nango/commit/11f2aa886c4abae2a42d5d20ac3a72e46f978154

In particular the lines: 
```
let headers: Record<string, string> = {};

if (template.token_request_auth_method === 'basic') {
    headers['Authorization'] = 'Basic ' + Buffer.from(config.oauth_client_id + ':' + config.oauth_client_secret).toString('base64');
}
```